x-pack/plugins/apm/public/components/routing/templates/apm_main_template.tsx

```diff
@@ -54,16 +54,26 @@ export function ApmMainTemplate({
     return callApmApi('GET /internal/apm/has_data');
   }, []);
 
+  const shouldBypassNoDataScreen = bypassNoDataScreenPaths.some((path) =>
+    location.pathname.includes(path)
+  );
+
+  const { data: fleetApmIntegrations } = useFetcher(
+    (callApmApi) => {
+      if (!data?.hasData && !shouldBypassNoDataScreen) {
+        return callApmApi('GET /internal/apm/fleet/has_data');
```

Comment:
The two `has_data` endpoints confused me at first. Can we rename this `has_policies`?

```suggestion
        return callApmApi('GET /internal/apm/fleet/has_policies');
```

Reply:
Changed the route in https://github.com/elastic/kibana/pull/126146/commits/6d18c704e209c04155d0430b197d3c958741154d